yb-voyager/src/tgtdb/oracle.go

```diff
@@ -0,0 +1,478 @@
+			if err2 != nil {
+				rowsAffected = 0
+				err = fmt.Errorf("commit transaction: %w", err2)
+			}
+		}
+	}()
+
+	var alreadyImported bool
+	alreadyImported, rowsAffected, err = db.isBatchAlreadyImported(tx, batch)
+	if err != nil {
+		return 0, err
+	}
+	if alreadyImported {
+		return rowsAffected, nil
+	}
+
+	// fmt.Printf("Importing table path: %s \n", batch.GetFilePath())
+	tableName := batch.GetTableName()
+
+	sqlldrControlFileContent := args.GetSqlLdrControlFile(db.tconf.Schema)
+
+	if _, err := os.Stat(fmt.Sprintf("%s/sqlldr", exportDir)); os.IsNotExist(err) {
+		os.Mkdir(fmt.Sprintf("%s/sqlldr", exportDir), 0755)
+	}
+	sqlldrControlFileName := fmt.Sprintf("%s.ctl", tableName)
+	sqlldrControlFilePath := fmt.Sprintf("%s/sqlldr/%s", exportDir, sqlldrControlFileName)
+	sqlldrControlFile, err := os.Create(sqlldrControlFilePath)
+	if err != nil {
+		return 0, fmt.Errorf("create sqlldr control file %q: %w", sqlldrControlFilePath, err)
+	}
+	defer sqlldrControlFile.Close()
+	_, err = sqlldrControlFile.WriteString(sqlldrControlFileContent)
+	if err != nil {
+		return 0, fmt.Errorf("write sqlldr control file %q: %w", sqlldrControlFilePath, err)
+	}
+
+	sqlldrLogFileName := fmt.Sprintf("%s.log", tableName)
+	sqlldrLogFilePath := fmt.Sprintf("%s/sqlldr/%s", exportDir, sqlldrLogFileName)
+	sqlldrLogFile, err := os.Create(sqlldrLogFilePath)
+	if err != nil {
+		return 0, fmt.Errorf("create sqlldr log file %q: %w", sqlldrLogFilePath, err)
+	}
+	defer sqlldrLogFile.Close()
+
+	// fmt.Println("Running sqlldr for file: ", sqlldrControlFilePath)
+	connectionString := db.getConnectionUri(db.tconf)
+	user := getValue(connectionString, "user")
+	password := getValue(connectionString, "password")
+	connectString := getValue(connectionString, "connectString")
+
+	oracleConnectionString := fmt.Sprintf("%s/%s@\"%s\"", user, password, connectString)
+
+	sqlldrArgs := fmt.Sprintf("userid=%s control=%s log=%s DIRECT=TRUE NO_INDEX_ERRORS=TRUE", oracleConnectionString, sqlldrControlFilePath, sqlldrLogFilePath)
+	// fmt.Println("Args: ", sqlldrArgs)
+
+	cmd := exec.Command("sqlldr", sqlldrArgs)
+	var outbuf bytes.Buffer
+	var errbuf bytes.Buffer
+	cmd.Stdout = &outbuf
+	cmd.Stderr = &errbuf
+	// fmt.Printf("cmd: %v\n", cmd)
+	err = cmd.Run()
+	log.Infof("sqlldr output: %s", outbuf.String())
+	log.Errorf("sqlldr error: %s", errbuf.String())
+	var err2 error
+	rowsAffected, err2 = getRowsAffected(outbuf.String())
+	if err2 != nil {
+		return 0, fmt.Errorf("get rows affected from sqlldr output: %w", err)
+	}
+
+	if err != nil {
+		// find ORA-00001: unique constraint * violated in log file
+		pattern := regexp.MustCompile(`ORA-00001: unique constraint \(.+?\) violated`)
+		scanner := bufio.NewScanner(sqlldrLogFile)
+		for scanner.Scan() {
+			line := scanner.Text()
+			if pattern.MatchString(line) {
+				err = db.recordEntryInDB(tx, batch, rowsAffected)
+				if err != nil {
+					err = fmt.Errorf("record entry in DB for batch %q: %w", batch.GetFilePath(), err)
+				}
```

Comment:
Instead of duplicating this code snippet, just set a flag `ignoreError` if a line matching the pattern exists. At line 370, you should return error only `if !ignoreError`.

Reply:
Done